src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -143,7 +143,11 @@ private function handleArrayNode(ArrayNode $node, ClassBuilder $class, string $n
         );
         $body = $hasNormalizationClosures ? '
 /**
+ * @template T
+ * @psalm-param T $value
+ *
  * @return CLASS|$this
+ * @psalm-return (T is array ? CLASS : static)
```

Comment:
shouldn't this be `$this` rather than `static` for consistency ?

Reply:
Psalm does not seem to support `$this` in conditional types: @psalm-return (T is array ? CLASS : static)

Reply:
Is that a bug? Should it be reported to psalm?

Reply:
Created a bug report: https://github.com/vimeo/psalm/issues/8348